### DISCUSSION TITLE
请教一下 nextjs 框架下版本下如何实现本地 proxy 的功能？

### DISCUSSION THREAD

**Discussion by arvinxx** [ddiu8081/chatgpt-demo#189] [Q&A] [answered]
首先我想对这个项目说一声太赞了！周日的时候眼看从几百星到现在3.7k星，势头太猛了~

我现在在使用nextjs  框架开发一个类似的项目，本地开发调试一个需要使用代理。想咨询下，在nextjs下应该如何实现类似的效果？

**Comment by ddiu8081** [comment, marked as answer]
对接口使用代理，如果使用 `undici` 等请求库进行请求，可以使用诸如 `dispatcher` 的参数为请求添加代理，参考 https://github.com/ddiu8081/chatgpt-demo/commit/3cc61371db524411775829073b4b5b40a5d10ff7 .

node 原生的 fetch 方法，似乎还不能设置代理。

Use a proxy for the request, if the request is made using a request library such as `undici`, you can add a proxy to the request using a parameter such as `dispatcher`

The node native fetch method does not seem to be able to set a proxy yet.

**Reply by arvinxx** [reply]
我也在尝试使用 undici ，但发现vercel的 Edge Runtime 似乎是不支持 node相关的接口的。当我配置了 undici 的 proxy，会返回 undefined of node 的报错。这个有遇到过么？

**Reply by ddiu8081** [reply]
Vercel Runtime 只能直接使用 fetch API，需要在部署到 Vercel 时移除 node 的引用。https://edge-runtime.vercel.app/features/available-apis

**Reply by arvinxx** [reply]
这个我知道的，我现在遇到的问题是：本地启动的时候，使用 undici 会报错（似乎在本地 vercel 也模拟了一个edge runtime ？），想知道你是怎么解决这个问题的~

**Comment by VincentHan888** [comment]
May I ask if use this kind of 'http-proxy-middleware' in this project, can make users free of proxy?https://github.com/chimurai/http-proxy-middleware @ddiu8081
